### PR TITLE
humanStructure overhaul 

### PR DESCRIPTION
**Pull request by vtikha**
This PR changes the ```humanStructure``` to include various enhancements of which :

1. Verify and **order** skeletons (left to right)
1. Select which skeleton to **track** (by lifting the arm)
1. Segment **Region of Interest** of the selected skeleton
1. Send a **trigger** to the speech pipeline in order to **start** and **stop** the **speech acquisition**. This is activated by **lifting the arm**, which sends a start trigger, keeping it up until the speech request is done, and **putting it back down** again to send the stop trigger

### CODE REVIEW

File: modules/humanStructure/CMakeLists.txt [hunk 1]
```diff
@@ -1,12 +1,20 @@
-# Copyright: (C) 2018 iCub Facility - Istituto Italiano di Tecnologia
+# Copyright: (C) 2019 iCub Facility - Istituto Italiano di Tecnologia
 # Authors: Vadim Tikhanoff
 # CopyPolicy: Released under the terms of the GNU GPL v2.0.
 
 project(humanStructure)
 
+message("YARP_LIBRARIES = ${YARP_LIBRARIES}")
```
Comment (pattacini) [inline modules/humanStructure/CMakeLists.txt]:
```suggestion
message(STATUS "YARP_LIBRARIES = ${YARP_LIBRARIES}")
```

File: modules/humanStructure/CMakeLists.txt [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+foreach(target ${YARP_LIBRARIES})
+    get_property(x TARGET ${target} PROPERTY INTERFACE_LINK_LIBRARIES)
+    message("    ${target} -> ${x}")
```
Comment (pattacini) [inline modules/humanStructure/CMakeLists.txt]:
```suggestion
    message(STATUS "    ${target} -> ${x}")
```

File: modules/humanStructure/CMakeLists.txt [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+endforeach()
+
 set(doc_files ${PROJECT_NAME}.xml)
 source_group("DOC Files" FILES ${doc_files})
 
+include_directories(${OpenCV_INCLUDE_DIRS})
```
Comment (pattacini) [inline modules/humanStructure/CMakeLists.txt]: You can safely remove this line, as `target_link_libraries` takes care of that.

File: modules/humanStructure/CMakeLists.txt [hunk 4]
```diff
@@ -1,12 +1,20 @@
...
+
```
Comment (pattacini) [inline modules/humanStructure/CMakeLists.txt]: The following line is required on Windows:
```cmake
target_compile_definitions(${PROJECT_NAME} PRIVATE _USE_MATH_DEFINES)
```

File: modules/humanStructure/main.cpp [hunk 5]
```diff
@@ -34,6 +37,8 @@
 #include <iostream>
 #include <utility>
```
Comment (pattacini) [inline modules/humanStructure/main.cpp]: We need to include:
```c++
#include <cmath>
```

File: modules/humanStructure/main.cpp [hunk 6]
```diff
@@ -42,8 +47,28 @@ class Processing : public yarp::os::BufferedPort<yarp::os::Bottle>
...
+    cv::Mat overlayObject;
+
+    bool    camera_configured;
+    double  fov_h;
+    double  fov_v;
+
+    double minVal;
+    double maxVal;
+    
+    double isArmLifted;
```
Comment (vvasco) [inline modules/humanStructure/main.cpp]: `isArmLifted` should be a bool?

File: modules/humanStructure/main.cpp [hunk 7]
```diff
@@ -42,8 +47,28 @@ class Processing : public yarp::os::BufferedPort<yarp::os::Bottle>
     yarp::os::RpcServer handlerPort;
     yarp::os::BufferedPort<yarp::sig::ImageOf<yarp::sig::PixelRgb> >    imageInPort;
     yarp::os::BufferedPort<yarp::sig::ImageOf<yarp::sig::PixelRgb> >    imageOutPort;
+    yarp::os::BufferedPort<yarp::sig::ImageOf<yarp::sig::PixelMono> >   imageOutDepthPort;
+    yarp::os::BufferedPort<yarp::sig::ImageOf<yarp::sig::PixelMono> >   imageOutSegmentPort;
+    yarp::os::BufferedPort<yarp::sig::ImageOf<yarp::sig::PixelFloat>>   imageInFloat;
     yarp::os::BufferedPort<yarp::os::Bottle >    targetPort;
+    yarp::os::BufferedPort<yarp::os::Bottle >    armPort;
```
Comment (vvasco) [inline modules/humanStructure/main.cpp]: Maybe this `armPort` should be a `yarp::os::RpcClient`, as we typically send the `start` command to the rpc port of `googleSpeech`.

Comment (vtikha) [reply modules/humanStructure/main.cpp]: Yes, this was what we agreed

File: modules/humanStructure/main.cpp (hunk 6 above)
Comment (vtikha) [reply modules/humanStructure/main.cpp]: indeed

File: modules/humanStructure/CMakeLists.txt (hunk 2 above)
Comment (vtikha) [reply modules/humanStructure/CMakeLists.txt]: ok will remove all this debug

File: modules/humanStructure/CMakeLists.txt (hunk 3 above)
Comment (vtikha) [reply modules/humanStructure/CMakeLists.txt]: got it

File: modules/humanStructure/CMakeLists.txt (hunk 1 above)
Comment (vtikha) [reply modules/humanStructure/CMakeLists.txt]: ok will remove all this debug

**Comment by vvasco** [comment]
After a chat f2f with @vtikha , we thought it would be good to add a service to change online the threshold used to find is the arm is lifted.

**Pull request merged**